File: modules/expo-context-menu/ios/ExpoContextMenuModule.swift
import ExpoModulesCore

public class ExpoContextMenuModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoContextMenu")

    View(ExpoContextMenuView.self) {
      Events("onPressMenuItem")

      Prop("menuConfig") { (view: ExpoContextMenuView, menuConfig: [String: Any]) in
        view.menuConfig = menuConfig
      }

      Prop("triggerOnLongPress") { (view: ExpoContextMenuView, triggerOnLongPress: Bool) in
        view.triggerOnLongPress = triggerOnLongPress
      }      
    }
  }
}

File: modules/expo-context-menu/android/src/main/java/expo/modules/contextmenu/ExpoContextMenuModule.kt
package expo.modules.contextmenu

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoContextMenuModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoContextMenu")

    View(ContextMenuView::class) {
      Events("onPressMenuItem")

      Prop("menuConfig") { view: ContextMenuView, menuConfig: Map<String, Any> ->
        view.menuConfig = menuConfig
      }

      Prop("triggerOnLongPress") { view: ContextMenuView, triggerOnLongPress: Boolean ->
        view.triggerOnLongPress = triggerOnLongPress
      }
    }
  }
}
